lua/treesitter-context.lua

```diff
@@ -653,7 +653,73 @@ local function normalize_node(node)
   return node
 end
 
-local function open(ctx_nodes)
+local virt_text_ns = api.nvim_create_namespace("nvim-treesitter-context-virt-text")
+---Render virtual text in the context buffer, includes extmarks and diagnostics
+---@param cbufnr integer buf number of the context buffer
+---@param extmarks table result of `clone_extmarks_into()`
+---@param diagnostics table result of `clone_diagnostics_into()`
+local function render_virtual_text(cbufnr, extmarks, diagnostics)
+  api.nvim_buf_clear_namespace(cbufnr, virt_text_ns, 0, -1)
+  vim.diagnostic.reset(virt_text_ns, cbufnr)
+
+  local len = api.nvim_buf_line_count(cbufnr)
+  for line = 0, len do
+    for _, m_info in ipairs(extmarks[line] or {}) do
+      api.nvim_buf_set_extmark(cbufnr, virt_text_ns, line, m_info.col, m_info.opts)
+    end
+  end
+
+  vim.diagnostic.set(virt_text_ns, cbufnr, diagnostics, { signs = false })
```

Comment:
I don't think this is necessary. The context buffer is just an implementation detail. Users aren't expected to interact with it. If absolute virtual_lines was a thing, we could use that instead.

Reply:
It's nice to have the consistency of setting the virtual text in the context buffer. Adding diagnostics to the context buffer is the easiest way to do that. 

Would you prefer that I just show the diagnostic message as virtual text?

Reply:
You're already copying all the extmarks (which includes all the diagnostic extmarks) so there's not much point copying the diagnostics too. You really only want to copy one or the other.

For all intents and purposes, the context buffer should be hidden from the user. We don't want to be setting false diagnostics there.

Reply:
Interesting, before the diagnostic copying there was no diagnostic virtual text showing in the context buffer. I can check again, but I'm guessing that diagnostic virtual text isn't namespaced. 

So I would need to reproduce the virtual text myself to get it to show up in the context buffer.

Reply:
It should: https://github.com/neovim/neovim/blob/7963a160e90f9ded63db1dfa24e607ee70af18ba/runtime/lua/vim/diagnostic.lua#L1010